python.d/tor.chart.py

```diff
@@ -0,0 +1,128 @@
+    STEM_AVAILABLE = False
+
+NAME = "Tor"
+NETDATA_CONFIG_DIR = os.environ.get('NETDATA_CONFIG_DIR', ".")
+
+# default module values
+# update_every = 10
+priority = 90000
+retries = 60
+
+ORDER = ['traffic']
+CHARTS = {
+    'traffic': {
+        'options': [None, "Tor traffic", 'bytes/s', 'Tor traffic',
+                    'tor.traffic', 'line'],
+        'lines': [
+            ["read", "read", "incremental"],
+            ["write", "write", "incremental"],
+        ]
+    }
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+
+        try:
+            if not STEM_AVAILABLE:
+                raise Exception("The stem library is missing")
+            self._conf = self.load_conf()
+            self._authenticate()
+            self._ready = True
+        except Exception as e:
+            self.error(str(e))
+            self._ready = False
+
+    def load_conf(self):
+        fn = os.path.join(NETDATA_CONFIG_DIR, "tor.conf")
+        with open(fn) as f:
+            j = json.load(f)
+        return j
+
+    def _authenticate(self):
+        connect_method = self._conf.get('connect_method', 'port')
+        password = self._conf.get('password', None)
+        if connect_method == 'port':
+            port = self._conf.get('port', 9051)
+            self._controller = stem.control.Controller.from_port(port=port)
+        else:
+            socket_path = self._conf.get('socket_path', '')
+            self._controller = stem.control.Controller.from_socket_file(
+                path=socket_path)
+
+        try:
+            self._controller.authenticate(password=password)
+        except stem.connection.AuthenticationFailure as e:
+            self.error('Authentication failed ({})'.format(e))
+
+    def check(self):
+        return self._ready
+
+    def get_metric(self, name):
+        try:
+            return self._controller.get_info(name)
+        except Exception:
+            # Try to recreate the controller and authenticate - just once
+            self._authenticate()
+            return self._controller.get_info(name)
+
+    def _get_data(self):
+        """Get host metrics
+        :return: dict
+        """
+        try:
+            tr = self.get_metric('traffic/read')
+            tw = self.get_metric('traffic/written')
+            data = dict(
+                read=int(tr),
+                write=int(tw)
+            )
+            return data
+        except Exception as e:
+            self.error("Error reading metrics: %s" % str(e))
+            return None
+
+
+if __name__ == '__main__' and NETDATA_CONFIG_DIR == ".":
+    # Running from CLI
+    s = Service(
+        name=NAME,
+        configuration=dict(
+            priority=priority,
+            retries=retries,
+            update_every=1
+        ),
+    )
+    s.update(1)
```

Comment:
No needed.

You can run python.d.plugin in debug mode.
https://github.com/firehol/netdata/wiki/Add-more-charts-to-netdata#configuring-plugins

```
/usr/libexec/netdata/plugins.d/python.d.plugin 1 debug mysql
```